tests/Tests/500__jupyterhub/test-jupyterhub-api-telemetry.robot

```diff
@@ -0,0 +1,78 @@
+Resource    ../../Resources/Page/ODH/JupyterHub/LoginJupyterHub.robot
+Resource    ../../Resources/Page/ODH/JupyterHub/JupyterHubSpawner.robot
+Resource    ../../Resources/Page/OCPDashboard/OCPMenu.robot
+Library     SeleniumLibrary
+
+*** Test Cases ***
+Verify Telemetry Data Is Accessible
+    [Tags]  Sanity
+    ...     ODS-347
+    Set Library Search Order  SeleniumLibrary
+    ${result} =   Is Usage Data Collection Enabled
+    Should Be True    ${result}
+    ${boolean} =    Checks SegmentKey Is Exposes In Jupyterhub API
+    Should Be True    ${boolean}
+    ${handle} =    Get Window Handles
+    Switch Window    ${handle}[0]
+    Sleep    2s    msg=Wait for to change tab
+    Navigate To Cluster Settings Page
+    Disable "Usage Data Collection"
+    Switch Window    ${handle}[1]
+    Reload Page
+    ${boolean} =    Check Is API Exposes SegmentKey
+    Should Not Be True    ${boolean}
+    [Teardown]    Teardown For Checks SegmentKey Is Exposes In Jupyterhub API TestCase    ${handle}
+
+*** Keywords ***
+Navigate To Cluster Settings Page
+    [Documentation]     Navigate to Cluste Settings page
+    Menu.Navigate To Page    Settings    Cluster settings
+    Wait Until Page Contains    Usage Data Collection
+
+Teardown For Checks SegmentKey Is Exposes In Jupyterhub API TestCase
+    [Documentation]     Enables ussage data collection and close the browser
+    [Arguments]     ${handle}
+    Switch Window    ${handle}[0]
+    Wait Until Page Contains    Usage Data Collection
+    Enable "Usage Data Collection"
+    ${result} =   Is Usage Data Collection Enabled
+    Should Be True    ${result}
+    Close Browser
+
+Checks SegmentKey Is Exposes In Jupyterhub API
+    [Documentation]     Loads JupyterHub API and checks is API exposes segmentkey
+    Load JupyterHub API Page
+    ${return_value} =    Check Is API Exposes SegmentKey
+    [Return]    ${return_value}
+
+Load JupyterHub API Page
+    [Documentation]     Loads Jupyterhub and navigate to API url
+    Open Browser    ${ODH_DASHBOARD_URL}    browser=${BROWSER.NAME}    options=${BROWSER.OPTIONS}
+    Login To RHODS Dashboard    ${TEST_USER.USERNAME}    ${TEST_USER.PASSWORD}    ${TEST_USER.AUTH_TYPE}
+    Wait for RHODS Dashboard to Load
+    ${version-check} =    Is RHODS Version Greater Or Equal Than    1.4.0
+    IF    ${version-check}==True
+        Launch JupyterHub From RHODS Dashboard Link
+    ELSE
+        Launch JupyterHub From RHODS Dashboard Dropdown
+    END
+    Login To Jupyterhub    ${TEST_USER.USERNAME}    ${TEST_USER.PASSWORD}    ${TEST_USER.AUTH_TYPE}
+    Page Should Not Contain    403 : Forbidden
+    ${authorization_required} =    Is Service Account Authorization Required
+    Run Keyword If    ${authorization_required}    Authorize jupyterhub service account
+    ${url} =    Get Location
+    Log    ${url}
+    ${url} =    Replace String    ${url}    /hub/spawn    /services/jsp-api/api/instance
+    Go To    ${url}
+    sleep  2s
```

Comment:
is it really necessary? could this be decreased to 0.5s for example

Reply:
sure